### PR TITLE
url-segments.ts update

### PR DESCRIPTION
**Pull request by nnguyen144**
### Category
- [ ] **Feature**
- [x] **Bug Fix**
- [ ] **Testing**
- [ ] **Documentation**
- [ ] **Other**

### Domain
- [X] **Frontend**
- [ ] **Backend**
- [ ] **Both**

### Description
url-segments.ts is updated to include new url segments for API calls for Events, Users, Hubs, and potentially more in the future.

### Related Issue(s)
[https://github.com/UNLV-CS472-672/2024-F-GROUP3-RebelHub/issues/57](https://github.com/UNLV-CS472-672/2024-F-GROUP3-RebelHub/issues/57)

### Modifications

- **url-segments.ts:** Updated to include new url segments for API calls for Events, Users, and Hubs
- **Any file that used an API call for Events, Users, and Hubs:** Updated to use the new functions from url-segments.ts instead of writing out the url in the string in the api call.
- **fetchPrivileges.js:" Updated to not include duplicate hub entries if a user is accidentally both an owner and moderator of a hub.
- Any previous changes from this PR: [https://github.com/UNLV-CS472-672/2024-F-GROUP3-RebelHub/pull/48](https://github.com/UNLV-CS472-672/2024-F-GROUP3-RebelHub/pull/48)

### Usage and Integration
All future API calls should use a function from urls-segments.ts, and new functions can be created in that file to accommodate new URL segments.

### Testing
The changes were simply tested by using the pages of the website and tests if the functionality is the same, as it should not change from this update.

### Mentions
These changes are useful for all developers in this group.

### CODE REVIEW

File: frontend/src/utils/url-segments.ts [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+    // #region Events
+    EVENTS_API_BASE: "api/events/",
+    EVENTS_CREATE: "create/",
+    EVENTS_DELETE: "/delete/",
+    EVENTS_UPDATE: "/update/",
+    // #endregion
+};
+
+// #region Functions to return a URL for an API call (hubs)
+
```
Comment (barrioskevin) [inline frontend/src/utils/url-segments.ts:62]: // "https://localhost:8000/api/hubs/"

**Comment by barrioskevin** [comment]
Hi! 
If we are going to use the URL segments file throughout the enitre project then we should make sure that anyone who works on anything in the front end also uses the same scheme.

You should have explained more in the usage section how the url segments file is meant to be used, imports etc.. And enforce that we all use it. If we are inconsistent when using this then our project is potentially more vulnerable and will look messy.

I like the fact that everything is in one file so that makes managing urls a little easier but if every function we are exporting is just a url then I would suggest a better name. We have names that begin with get, and then include a create. If everything is a url it's prolly unnecessary to include url in all the names. But they way this file is setup I can see why it's necessary. 

I would have preferred exporting a constant object holding each url and we can access it as such `URL.HubsList` but thats just me nitpicking. 

The only thing i would suggest is adding comments above each function showing the entire url of the backend call. That way it's clear the exact endpoint the function is calling. And if we make edits in the backend we can quickly come to this file and see if anything needs to be added, removed, or updated.

Thank you for your time!

**Review by thecatfour — APPROVED** [review]
These changes should be good. In the future, we may want to create a comment in the file to demonstrate how the functions are used, but I believe that looking at them gives a good enough idea on what it's doing. Specifying that "get..." is for communicating with the backend and "goto..." is for redirecting in the fronted may be a good idea.

Regarding @imawful's comment, we could create a separate issue to improve documentation for the URL segments file. Since this PR and issue are more about updating every file to the new URL segments file, I don't think significantly detailed documentation is necessary.

**Comment by barrioskevin** [comment]
I think the best time to add in documentation to url segments is now. We don't have any testing framework for the frontend so unless you manually checked each and every function call to url segments it can be difficult to ensure that all links are working properly. If there is at least a comment above each function with the correct url we can at least know what it SHOULD be in the case it ever fails. 

If you don't want to add those comments in now, thats fine. It's just hard to approve of this with out any sort of quality assurance. But that might be a greater scope than this PR.

**Review by barrioskevin — APPROVED** [review]
The updated comments to URL segments look great. Thank you for taking the time to go through and update all the files. This passes the ci test so things should be ready to merge!

**Pull request merged**